[ENHANCEMENT] Switch Clans Without Exiting to Desktop

Treefire33 opened the pull request:
## About The Pull Request
Implements a way for the game to switch clans without having to close the game.
Also adds a way to "hot reload" a clan with F5.

## Linked Issues
closes: #3067 

## Proof of Testing
https://github.com/user-attachments/assets/c056f106-7e80-4658-84c2-63dccb8f5ff2

(Note: proof of testing shows some bugs, but they aren't detrimental to the game running.)

Treefire33 committed:
It works!

would solve #3067 , however I don't know if it's the best, or even good for that matter.

Treefire33 committed:
It works!

would solve #3067 , however I don't know if it's the best, or even good for that matter.

Treefire33 committed:
Merge branch 'ClanGenOfficial:development' into switchnoclose

Treefire33 committed:
improved and runs fine

Treefire33 committed:
improved and runs fine

Treefire33 committed:
Merge branch 'switchnoclose' of https://github.com/Treefire33/clangen into switchnoclose

Treefire33 committed:
some minor improvements

clears patrol things and completely empties the events list

Treefire33 committed:
leader den screen fix

also undid the change in utility

Treefire33 committed:
fix frame-long switch clan screen when switching

Treefire33 committed:
add keybind to reload a clan

Treefire33 committed:
update SwitchClanScreen note

j-gynn commented:
Was saving always this non-performant for you (looking at the video, this seems like an abnormally long save time). It also seems like the game is being hard-reset every time, instead of only reloading the relevant data (the libpng warning is displayed again, which is probably contributing to the slow load times).

I am personally a little concerned with having a debug tool that could potentially lose players save data accessible from a simple mispress. I don't know whether a kind of "sudo" mode could be implemented for dev tools so that regular players can't accidentally trigger game-wrecking features (perhaps in the same vein as `show_bounds` and co. that must be toggled using the debug console or via changing a section of the code).

I'm super stoked to see this feature added & I know that players are gonna love it. It's a huge QoL that's been on my hit list for months so I'm glad someone's finally taking a stab at it.

Treefire33 committed:
remove reload keybind and other things

Treefire33 committed:
Merge branch 'development' into switchnoclose

Treefire33 committed:
Reload clan without saving

Treefire33 committed:
fix some bugs

Treefire33 commented:
Okay, seems that the two issues listed might have been caused by the inclusion of loading all the spritesheets again.

Treefire33 committed:
Merge branch 'ClanGenOfficial:development' into switchnoclose

Treefire33 committed:
some changes

Mainly just changing how some reused lists are created more than once.

Treefire33 committed:
clan reload is back

It's a debug command this time around, and it also allows the user to save the clan before reloading.

Treefire33 commented:
I added a command, so here's some testing for it:

https://github.com/user-attachments/assets/790e1d8d-626f-4e87-9fcd-4391f1e3a3ae

Treefire33 committed:
Merge branch 'ClanGenOfficial:development' into switchnoclose

Treefire33 committed:
Merge branch 'ClanGenOfficial:development' into switchnoclose

j-gynn committed:
Merge branch 'development' into switchnoclose
